cmd/sriov-network-config-daemon/service.go

```diff
@@ -102,105 +102,155 @@ func runServiceCmd(cmd *cobra.Command, args []string) error {
+			return nil, nil, fmt.Errorf("sriov-config-service: failed to create virtual plugin %v", err)
 		}
 	}
-
-	nodeState := &sriovv1.SriovNetworkNodeState{
-		Spec:   nodeStateSpec.Spec,
+	return configPlugin, &sriovv1.SriovNetworkNodeState{
+		Spec:   conf.Spec,
 		Status: sriovv1.SriovNetworkNodeStateStatus{Interfaces: ifaceStatuses},
-	}
+	}, nil
+}
 
-	_, _, err = configPlugin.OnNodeStateChange(nodeState)
+func handleResultFile(setupLog logr.Logger) error {
+	if phase == PhasePre {
+		// make sure there is no stale result file to avoid situation when we
+		// read outdated info in the Post phase when the Pre silently failed (should not happen)
+		if err := systemd.RemoveSriovResult(); err != nil {
+			return fmt.Errorf("failed to remove result file: %v", err)
+		}
+		return nil
+	}
+	// we are in the Post phase, check result file created by the Pre phase
+	setupLog.V(0).Info("check result of the Pre phase")
+	prePhaseResult, err := systemd.ReadSriovResult()
 	if err != nil {
-		setupLog.Error(err, "failed to run OnNodeStateChange to update the generic plugin status")
-		return updateSriovResultErr(fmt.Sprintf("sriov-config-service: failed to run OnNodeStateChange to update the generic plugin status %v", err))
+		return fmt.Errorf("failed to read result of the pre phase: %v", err)
 	}
-
-	sriovResult := &systemd.SriovResult{
-		SyncStatus:    consts.SyncStatusSucceeded,
-		LastSyncError: "",
+	if prePhaseResult.SyncStatus != consts.SyncStatusInProgress {
+		return fmt.Errorf("unexpected result of the pre phase: %s, syncError: %s", prePhaseResult.SyncStatus, prePhaseResult.LastSyncError)
 	}
+	setupLog.V(0).Info("Pre phase succeed, continue execution")
+	return nil
+}
 
-	err = configPlugin.Apply()
+func updateSriovResultErr(setupLog logr.Logger, origErr error) error {
+	setupLog.Error(origErr, "service call failed")
+	err := updateResult(setupLog, consts.SyncStatusFailed, fmt.Sprintf("%s: %s", phase, origErr.Error()))
 	if err != nil {
-		setupLog.Error(err, "failed to run apply node configuration")
-		sriovResult.SyncStatus = consts.SyncStatusFailed
-		sriovResult.LastSyncError = err.Error()
+		return err
 	}
+	return origErr
+}
 
-	err = systemd.WriteSriovResult(sriovResult)
-	if err != nil {
-		setupLog.Error(err, "failed to write sriov result file", "content", *sriovResult)
-		return fmt.Errorf("sriov-config-service failed to write sriov result file with content %v error: %v", *sriovResult, err)
+func updateSriovResultSucceed(setupLog logr.Logger) error {
+	setupLog.V(0).Info("service call succeed")
+	syncStatus := consts.SyncStatusSucceeded
+	if phase == PhasePre {
+		syncStatus = consts.SyncStatusInProgress
 	}
-
-	setupLog.V(0).Info("shutting down sriov-config-service")
-	return nil
+	return updateResult(setupLog, syncStatus, "")
 }
 
-func updateSriovResultErr(errMsg string) error {
+func updateResult(setupLog logr.Logger, result, msg string) error {
```

Comment:
> that way we dont need that in progress on pre phase at L238.

I think this is a nice thing that we set SyncStatus to InProgress in the "pre" phase because:
1. We are really in the process of the configuration at this point, so the status reflects the actual state.
2. The logic in the daemon and in the service is very simple and you can describe it with the single sentence. The daemon is not aware about phases at all.

Reply:
ack, if it will get more complex down the road and find we need more information about the outcome of each state (e.g for config daemon), we can extend.